aws/logs_monitoring/parsing.py

```diff
@@ -527,6 +528,47 @@ def cwevent_handler(event, metadata):
     yield data
 
 
+# Extract findings from Security Hub logs
+# and move them to the root of the log
+def separate_security_hub_findings(event):
+    if (
+        event.get(DD_SOURCE) == "securityhub"
+        and event.get("detail", {}).get("findings") is not None
+    ):
+        events = []
+        event_copy = copy.deepcopy(event)
+        # Copy findings before separating
+        findings = event_copy.get("detail", {}).get("findings")
+        if findings:
+            # Remove findings from the original event once we have a copy
+            del event_copy["detail"]["findings"]
+            # For each finding create a separate log event
+            for index, item in enumerate(findings):
+                # Copy the original event with source and other metadata
+                new_event = copy.deepcopy(event_copy)
+                current_finding = findings[index]
+                # Get the resources array from the current finding
+                resources = current_finding.get("Resources", {})
+                new_event["detail"]["finding"] = current_finding
+                new_event["detail"]["finding"]["resources"] = {}
+                # Separate objects in resources array into distinct attributes
+                if resources:
+                    # Remove from current finding once we have a copy
+                    del current_finding["Resources"]
+                    for item in resources:
+                        current_resource = item
+                        # Capture the type and use it as the distinguishing key
+                        resource_type = current_resource.get("Type", {})
+                        del current_resource["Type"]
+                        new_event["detail"]["finding"]["resources"][
+                            resource_type
+                        ] = current_resource
+                events.append(new_event)
+        return events
+    else:
```

Comment:
By default python functions return `None`, so this else is redundant. If you do prefer explicit returns (which is a good practice), the common practice is to do an early return to make the code easier to read.

```python
if event.get(DD_SOURCE) != "securityhub" or not event.get("detail", {}).get("findings"):
    return None

events = []
event_copy = copy.deepcopy(event)
... 
```

This also helps reduce the indentation, which makes the code easier to read.

Reply:
Big +1 on this comment -- let's see if we can reduce the indentation here by returning early